src/Select/Select.stories.tsx

```diff
@@ -163,111 +147,28 @@ class MultiSelectCheckbox extends React.Component<Props, State> {
     selectedOptions: [],
   };
 
-  handleOceanClick() {
-    const redOption = colorOptions.find(o => o.value === 'red');
-    redOption.isDisabled = !redOption.isDisabled;
-  }
-
-  handleCheck(val) {
-    if (val === 'ocean') {
-      this.handleOceanClick();
-    }
-
-    const isSelectedOption = this.state.selectedOptions.find(
-      o => o.value === val,
-    );
-    let res: any[] = [];
-    if (isSelectedOption) {
-      res = this.state.selectedOptions.filter(option => option.value !== val);
-    } else {
-      res = [
-        ...this.state.selectedOptions,
-        colorOptions.find(o => o.value === val),
-      ];
-    }
-    this.setState(state => ({
-      selectedOptions: res,
-    }));
-  }
-
   handleSelect(e) {
-    const newOceanOption = e.find(o => o.value === 'ocean');
-    const oldOceanOption = this.state.selectedOptions.find(
-      o => o.value === 'ocean',
-    );
-    if (
-      (newOceanOption && !oldOceanOption) ||
-      (!newOceanOption && oldOceanOption)
-    ) {
-      this.handleOceanClick();
-    }
+    const selected = Array.isArray(e) ? e : [e];
     this.setState(state => ({
-      selectedOptions: e,
+      selectedOptions: selected,
     }));
   }
 
-  MenuList = (props: any) => {
-    return (
-      <>
-        <div className={'menuListHeader'}>
-          {this.state.selectedOptions.length} items selected
-        </div>
-        {props.children}
-      </>
-    );
-  };
-
-  MultiValue = (props: any) => {
-    const withComma = `${props.data.label}, `;
-    return (
-      <div>
-        {this.state.selectedOptions[this.state.selectedOptions.length - 1]
-          .value !== props.data.value ? (
-          <SDiv>{withComma} &#32;</SDiv>
-        ) : (
-          <SDiv>{props.data.label}</SDiv>
-        )}
-      </div>
-    );
-  };
-
-  Option = (props: any) => {
-    return (
-      <SelectComponents.Option {...props}>
-        <SCheckbox
-          id={props.value}
-          defaultChecked={props.isSelected}
-          checked={props.isSelected}
-          disabled={props.isDisabled}
-          value={props.value}
-          onChange={() => this.handleCheck(props.value)}
-        >
-          <span>{props.data.label}</span>
-        </SCheckbox>
-      </SelectComponents.Option>
-    );
-  };
-
   render() {
     return (
       <FlexDiv>
         <Select
           creatable={boolean('creatable', true)}
-          isMulti
-          components={{
-            Option: this.Option,
-            MultiValue: this.MultiValue,
-            MenuList: this.MenuList,
-          }}
+          isMulti={boolean('isMulti', true)}
+          optionType={'checkbox'}
```

Comment:
Please provide a knob to manage optionType value. In same way as selectSize

Reply:
will do, thanks

Reply:
fixed!